csharp/src/Apache.Arrow/Arrays/PrimitiveArray.cs

```diff
@@ -86,5 +86,29 @@ IEnumerator IEnumerable.GetEnumerator()
                 yield return IsValid(index) ? Values[index] : null;
             }
         }
+
+        int ICollection<T?>.Count => Length;
+        bool ICollection<T?>.IsReadOnly => true;
+        void ICollection<T?>.Add(T? item) => throw new NotSupportedException("Collection is read-only.");
+        bool ICollection<T?>.Remove(T? item) => throw new NotSupportedException("Collection is read-only.");
+        void ICollection<T?>.Clear() => throw new NotSupportedException("Collection is read-only.");
+
+        bool ICollection<T?>.Contains(T? item)
+        {
+            if (item == null)
+            {
+                return NullCount > 0;
+            }
+
+            return Values.IndexOf(item.Value) >= 0;
```

Comment:
This isn't sufficient because the data buffer has to be initialized with something but the individual value might still be null. Consider an int32 vector of all nulls. There's a pretty good probability that Values.IndexOf(0) would find something.

Reply:
Okay, I have added a test to catch this bug in my implementation, and switched to the implementation that you recommended.
I have only optimised `Contains` for `PrimitiveArray` for now.